File: AI.Core/OperatorsImplimentation/BasicOperators/Time.cs
using System;

namespace AI.Core.OperatorsImplimentation.BasicOperators
{
    class Time : Operator
    {
        /// <summary>
        /// Returns UTC ticks
        /// </summary>
        public Time(Entity thisEntity)
            : base(thisEntity)
        {
            EnterContactsCount = 0;
        }

        public override void Action()
        {
            ExitContacts.SetValue((ulong)DateTime.UtcNow.Ticks); 
        }
    }
}

File: NativeCore/OperatorsImplimentation/BasicOperators/Time.cpp
#include "Time.h"

//C# TO C++ CONVERTER TODO TASK: The .NET System namespace is not available from native C++:
//using namespace System;
namespace AI
{
	namespace Core
	{
		namespace OperatorsImplimentation
		{
			namespace BasicOperators
			{

				Time::Time(Entity *thisEntity) : Operator(thisEntity)
				{
					setEnterContactsCount(0);
				}

				void Time::Action()
				{
					ExitContacts->SetValue(static_cast<unsigned long long>(DateTime::UtcNow.Ticks));
				}
			}
		}
	}
}
